### ISSUE TITLE
Errors when we try to use the latest .onnx model provided in libfacedetection.train

### ISSUE THREAD

**Issue by yjump** [opencv/opencv_zoo#146]
Hi feng, 
Many thans for your useful work! When we try to use the latest .onnx model provided in libfacedetection.train, we find the following error, would you mind give us any advice how to use the latest .onnx model provided in libfacedetection.train?

os: Linux
c++ tool chain: gcc-linaro-7.5.0-2019.12-x86_64_aarch64-linux-gnu
opencv:4.7.0
cmd: /demo.cpp -m=yunet_s_320_320.onnx ( from https://github.com/ShiqiYu/libfacedetection.train/tree/master/onnx) -i=test2.png -s=true
error:
terminate called after throwing an instance of 'cv::Exception'
  what():  OpenCV(4.7.0) /usr1/opencv-4.7.0/modules/dnn/src/net_impl.cpp:275: error: (-204:Requested object was not found) Layer with requested id=-1 not found in function 'getLayerData'

**Comment by fengyuentau** [comment]
Duplicate of https://github.com/opencv/opencv_zoo/issues/81.

Long story short: the demo use `cv.FaceDetectorYN` to load and infer the model. This API has implementation of pre- and post-processings specific to a older version of YuNet. Latest YuNet has a lot of differences from the one used by `cv.FaceDetectorYN`. So it won't work.

I propose to use the demo and model we provide here in the zoo. Or you can wait a bit for the update of `cv.FaceDetectorYN` https://github.com/opencv/opencv/pull/23020.

By the way, I also tried with the basic OpenCV DNN APIs and they work with this model:

```shell
$ python
>>> import numpy as np
>>> import cv2 as cv
>>> cv.__version__
'4.7.0'
>>> net = cv.dnn.readNet("yunet_s_320_320.onnx")
>>> x = np.random.randn(1, 3, 320, 320)
>>> net.setInput(x)
>>> net.forward()
```

**Comment by yjump** [comment]
> Duplicate of #81.
> 
> Long story short: the demo use `cv.FaceDetectorYN` to load and infer the model. This API has implementation of pre- and post-processings specific to a older version of YuNet. Latest YuNet has a lot of differences from the one used by `cv.FaceDetectorYN`. So it won't work.
> 
> I propose to use the demo and model we provide here in the zoo. Or you can wait a bit for the update of `cv.FaceDetectorYN` [opencv/opencv#23020](https://github.com/opencv/opencv/pull/23020).
> 
> By the way, I also tried with the basic OpenCV DNN APIs and they work with this model:
> 
> ```shell
> $ python
> >>> import numpy as np
> >>> import cv2 as cv
> >>> cv.__version__
> '4.7.0'
> >>> net = cv.dnn.readNet("yunet_s_320_320.onnx")
> >>> x = np.random.randn(1, 3, 320, 320)
> >>> net.setInput(x)
> >>> net.forward()
> ```

Thanks a lot for you timely and kind reply. 
If I understand correctly, I can try to follow `cv.FaceDetectorYN` [opencv/opencv#23020](https://github.com/opencv/opencv/pull/23020) to load "yunet_s_320_320.onnx"  to implement C++ inference demo?

**Comment by fengyuentau** [comment]
> If I understand correctly, I can try to follow cv.FaceDetectorYN https://github.com/opencv/opencv/pull/23020 to load "yunet_s_320_320.onnx" to implement C++ inference demo?

Yes, but you will need to clone and build from that branch yourself for now.
